code/game/objects/structures/flora/plant.dm

```diff
@@ -0,0 +1,99 @@
+/obj/structure/flora/plant
+	icon = 'icons/obj/hydroponics/hydroponics_growing.dmi'
+	icon_state = "bush5-4"
+	color = COLOR_GREEN
+	var/growth_stage
+	var/dead = FALSE
+	var/sampled = FALSE
+	var/datum/seed/plant
+	var/harvestable
+
+/* Notes for future work moving logic off hydrotrays onto plants themselves:
+/obj/structure/flora/plant/Process()
+	// check our immediate environment
+	// ask our environment for available reagents
+	//    process the reagents
+	// ask our environment for gas
+	//    take gas/add gas from environment
+	// ask our environment for light
+	// update health
+	// advance age
+	// update icon/harvestability as appropriate
+*/
+
+/obj/structure/flora/plant/examine(mob/user, distance)
+	. = ..()
+	if(dead)
+	else if(length(harvestable))
+		to_chat(user, SPAN_NOTICE("You can see [length(harvestable)] harvestable fruit\s."))
+
+/obj/structure/flora/plant/Initialize(ml, _mat, _reinf_mat, datum/seed/_plant)
+	if(!plant && _plant)
+		plant = _plant
+	if(!plant)
+		return INITIALIZE_HINT_QDEL
+	name = plant.display_name
```

Comment:
Is the theory that subtypes might populate `plant` before calling `..()`? Otherwise hard to see how it can be an instance here except through the 4th arg method, in which case the block above wouldn't need the `if(!plant...` condition.